classes/01class/exercises/c01-aws02/nimmi89/README.md

```diff
@@ -0,0 +1,210 @@
+        "InstanceId": "i-05d7620963b672ca0",
+        "IamInstanceProfile": {
+            "Arn": "arn:aws:iam::438549961569:instance-profile/S3AccessRole-Instance-Profile",
+            "Id": "AIPAWMG45LNQ2WM5DWF5E"
+        },
+        "State": "associating"
+    }
+}
+
+>>>Command to verify that the IAM role is now attached to the instance
+aws ec2 describe-iam-instance-profile-associations
+>The command returns:
+{
+    "IamInstanceProfileAssociations": [
+        {
+            "AssociationId": "iip-assoc-0ad4a200b37494d0b",
+            "InstanceId": "i-03db9e088f065c42b",
+            "IamInstanceProfile": {
+                "Arn": "arn:aws:iam::438549961569:instance-profile/cloudwatchrole",
+                "Id": "AIPAWMG45LNQWFJASE25M"
+            },
+            "State": "associated"
+        },
+        {
+            "AssociationId": "iip-assoc-09051d5837a6bb5ba",
+            "InstanceId": "i-05d7620963b672ca0",
+            "IamInstanceProfile": {
+                "Arn": "arn:aws:iam::438549961569:instance-profile/S3AccessRole-Instance-Profile",
+                "Id": "AIPAWMG45LNQ2WM5DWF5E"
+            },
+            "State": "associated"
+        }
+    ]
+}
+>>>SSH into the instance and access your bucket
+ssh -i /home/nimmi/C01-AWS01KeyPair.pem ec2-user@52.64.28.243
+Last login: Wed Jun 17 08:29:16 2020 from 14-201-149-89.static.tpgi.com.au
+
+       __|  __|_  )
+       _|  (     /   Amazon Linux 2 AMI
+      ___|\___|___|
+
+https://aws.amazon.com/amazon-linux-2/
+4 package(s) needed for security, out of 8 available
+Run "sudo yum update" to apply all updates.
+[ec2-user@ip-172-31-32-188 ~]$ aws s3 ls
+2020-06-17 08:16:40 c01-aws02-bucket
+
+S3 Access Successful in the EC2 using IAM Roles!!!!!!!!!!!
+
+```
+- Commands to copy the S3 file to a folder inside the instance (executed from inside the EC2 Instance):
+```
+>>>Command to copy s3 file to a folder inside the instance
+aws s3 cp s3://c01-aws02-bucket/index.html myEc2Folder
+>The command returns:
+download: s3://c01-aws02-bucket/index.html to myEc2Folder/index.html
+
+S3 File Successfully copied into Ec2 folder!!!!!!!!!
+[ec2-user@ip-172-31-32-188 ~]$ cd myEc2Folder/
+[ec2-user@ip-172-31-32-188 myEc2Folder]$ ls
+index.html
+
+>>>Command to upload a file into S3 bucket from the instance
+Without the permissions, the action is not successful
+[ec2-user@ip-172-31-32-188 ~]$ aws s3 cp file.txt s3://c01-aws02-bucket/
+upload failed: ./file.txt to s3://c01-aws02-bucket/file.txt An error occurred (AccessDenied) when calling the PutObject operation: Access Denied
+[ec2-user@ip-172-31-32-188 ~]$ logout
+Connection to 54.252.231.101 closed.
+
+>>>Create a policy to upload object onto bucket[Giving only the required privileges]
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Sid": "VisualEditor0",
+            "Effect": "Allow",
+            "Action": "s3:PutObject",
+            "Resource": "*"
```

Comment:
you're giving the required privilege, but you're allowing this privilege to any bucket.
Would be better to restrict the bucket that this role will have access to. Can you make this improvement?

Reply:
Yes, I thought so initially. But then I thought I don't have many buckets anyway. Makes Perfect sense though!have made the changes!